dashboard/app/models/pd/application/teacher1819_application.rb

```diff
@@ -764,8 +764,12 @@ def get_first_selected_workshop
       # Map back to actual workshops by reconstructing the friendly_date_range
       workshops = Pd::Workshop.where(id: workshop_ids)
       hash[:able_to_attend_multiple].each do |response|
-        selected_workshop = workshops.find {|w| response.start_with?(w.friendly_date_range)}
-        return selected_workshop if selected_workshop
+        workshops_for_date = workshops.select {|w| response.start_with?(w.friendly_date_range)}
+        return workshops_for_date.first if workshops_for_date.size == 1
+
+        location = response.scan(/in (.+) hosted/).first.try(:first) || ''
+        workshops_for_date_and_location = workshops_for_date.find {|w| w.location_address == location} || workshops_for_date.first
+        return workshops_for_date_and_location if workshops_for_date_and_location
```

Comment:
nit: we can just return the result of line 771. No need to store it and explicitly return if it exists. 

note: the last lines of a Ruby method being this:
```ruby
value = some_expression
return value if value
```

Presuming, that `value` didn't have a prior value, is always equivalent to:
```ruby
some_expression
```

Reply:
I don't quite understand this - will follow up when you are in